scripts/ssm-build-deb.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Execute the following as root to install lintian and fpm
+# apt-get install lintian
+# apt-get install ruby ruby-dev rubygems build-essential
+# gem install --no-ri --no-rdoc fpm
+
+# Then run this file, altering the version.
+# This file will create two versions of the deb file:
+# - apel-ssm_<version>_all.deb contains all the files necessary to run a
+#   the SSM as a sender.
+# - apel-ssm-service_<version>_amd64.deb will install service files
+#   necessary to run the SSM as a receiver on amd64 style architectures.
+# After building apel-ssm_<version>_all.deb, this script will run it
+# against lintian to highlight potential issues to the builder.
+
+set -eu
+
+TAG=2.2.0-1
+
+SOURCE_DIR=~/debbuild/source
+BUILD_DIR=~/debbuild/build
+
+# Where to install the python lib files
+PYTHON_INSTALL_LIB=/usr/lib/python2.7/dist-packages
+
+# Split the tag into version and package number
+# so they can be passed to fpm separately.
+# This will work with RCs of the form X.Y.Z-0.A.rcA
+VERSION=`echo "$TAG" | cut -d - -f 1`
+ITERATION=`echo "$TAG" | cut -d - -f 2`
+
+# Create SSM and DEB dir (if not present)
+mkdir -p $SOURCE_DIR
+mkdir -p $BUILD_DIR
+
+# Clean up any previous build
+rm -rf $SOURCE_DIR/*
+rm -rf $BUILD_DIR/*
+
+# Get and extract the source
+TAR_FILE=${TAG}.tar.gz
+TAR_URL=https://github.com/apel/ssm/archive/$TAR_FILE
+wget --no-check-certificate $TAR_URL -O $TAR_FILE
+tar xvf $TAR_FILE -C $SOURCE_DIR
+rm -f $TAR_FILE
+
+fpm -s python -t deb \
+-n apel-ssm \
+-v $VERSION \
+--iteration $ITERATION \
+-m "Apel Administrators <apel-admins@stfc.ac.uk>" \
```

Comment:
There's a lot of config here that seems like it would be better handled with a config file or as constants earlier in the script. Email is repeated.

Can the `depends` be given as a list?

Reply:
It probably would be better as a config file, or at the very least reducing the duplication of the email.

I don't think so, the documentation around `depends` says `A dependency. This flag can be specified multiple times.`

Reply:
Hmm, is there an option to read a list of dependencies or anything like that?

Reply:
I don't think so. Ideally, dependencies would be read from the `setup.py` script but as some of the python packages don't have a system packages I didn't use that feature.

Reply:
How is best to do this as a config file? Sourcing a separate file that list the variables? Like this:

```
TAG=...
SOURCE_DIR=...
BUILD_DIR=...
# etc etc
```

?

Reply:
As there's no built in way, I'd leave it as is.